### PR TITLE
PR edit so new test passes

### PR DESCRIPTION
**Pull request by Cristian-Baeza**
- Edited the check_for_wins_or_blocks method to what Tyler suggested in last PR 
- Edited best_move method to pass Tyler's new test
Had to edit it so it prioritized winning based on the current_player over blocking the opposite player

### CODE REVIEW

File: lib/tic_tac_toe.rb [hunk 1]
```diff
@@ -59,14 +59,25 @@ def board_full?
...
     end
 
+
     #CPU LOGIC METHODS
+    def prioritize_player(player_to_win, player_to_block)
+        move = cpu_check_for_wins_or_blocks(player_to_win)
+        move ||= cpu_check_for_wins_or_blocks(player_to_block)
+        move ||= cpu_take_open_space()
+        move
+    end
```
Comment (td-extreme) [inline lib/tic_tac_toe.rb:79]:
```suggestion
    def best_move()
        move = cpu_check_for_wins_or_blocks(current_player())
        move ||= cpu_check_for_wins_or_blocks(non_current_player())
        move ||= cpu_take_open_space()
    end
```

File: lib/tic_tac_toe.rb [hunk 2]
```diff
@@ -59,14 +59,25 @@ def board_full?
...
+
```
Comment (td-extreme) [inline lib/tic_tac_toe.rb:80]:
```suggestion
    def non_current_player
        current_player == :player_one ? :player_two : :player_one
    end

```

File: lib/tic_tac_toe.rb [hunk 3]
```diff
@@ -59,14 +59,25 @@ def board_full?
...
-      move ||= cpu_take_open_space()
+      if current_player() == :player_one
+        move = prioritize_player(:player_one, :player_two)
+      else
+        move = prioritize_player(:player_two, :player_one)
+      end
       move
     end
 
+
```
Comment (td-extreme) [inline lib/tic_tac_toe.rb:72]:
```suggestion
    def non_current_player
        current_player == :player_one ? :player_two : :player_one
    end
```

**Review by td-extreme — COMMENTED** [review]
Instead of having an if else and then passing those into your new fn `prioritize_player(player_to_win, player_to_block)` 
 Why don't we used the best_move how it was, and replace `:player_one` and `:player_two` with `current_player()` and a new `non_current_player()` method?

File: lib/tic_tac_toe.rb [hunk 4]
```diff
@@ -59,14 +59,25 @@ def board_full?
...
     def best_move
-      move = cpu_check_for_wins_or_blocks(:player_two)
-      move ||= cpu_check_for_wins_or_blocks(:player_one)
-      move ||= cpu_take_open_space()
+      if current_player() == :player_one
+        move = prioritize_player(:player_one, :player_two)
+      else
+        move = prioritize_player(:player_two, :player_one)
+      end
       move
```
[the remark this answers is not in the document]
Comment (td-extreme) [reply lib/tic_tac_toe.rb:69]: You don't need to have the last statement here be `move`
The statement `move ||= cpu_take_open_space()` will return the value of `move` already.

**Comment by Cristian-Baeza** [comment]
> Instead of having an if else and then passing those into your new fn `prioritize_player(player_to_win, player_to_block)` Why don't we used the `best_move` how it was, and replace `:player_one` and `:player_two` with `current_player()` and a new `non_current_player()` method?

Ohhhhh ok, sounds good!

File: lib/tic_tac_toe.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (Cristian-Baeza) [reply lib/tic_tac_toe.rb:69]: 👍🏽

File: lib/tic_tac_toe.rb (hunk 2 above)
Comment (Cristian-Baeza) [reply lib/tic_tac_toe.rb:80]: 👍🏽

File: lib/tic_tac_toe.rb (hunk 3 above)
Comment (Cristian-Baeza) [reply lib/tic_tac_toe.rb:72]: 👍🏽

**Review by td-extreme — APPROVED** [review]

**Pull request merged**